api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -894,11 +895,24 @@ message ScopedRoutes {
         }
       }
 
+      // Specifies how dynamic metadata values should be extracted.
+      message DynamicMetadataValueExtractor {
+        oneof metadata_extract_type {
+          option (validate.required) = true;
+
+          // Extract the entire string value for the provided metadata key.
+          type.metadata.v3.MetadataKey metadata_key = 1;
+        }
+      }
```

Comment:
What other options do you foresee here? Trying to understand how extensible we want this section of the API to be.